go-controller/cmd/ovn-k8s-cni-overlay/app/helper_windows.go

```diff
@@ -0,0 +1,220 @@
+}
+
+// createHNSEndpoint creates the HNS endpoint with the given configuration.
+// On success it returns the created HNS endpoint.
+func createHNSEndpoint(hnsConfiguration *hcsshim.HNSEndpoint) (*hcsshim.HNSEndpoint, error) {
+	logrus.Infof("Creating HNS endpoint")
+	hnsConfigBytes, err := json.Marshal(hnsConfiguration)
+	if err != nil {
+		logrus.Errorf("Could not create the HNSEndpoint, error: %v", err)
+		return nil, err
+	}
+	logrus.Infof("hnsConfigBytes: %v", string(hnsConfigBytes))
+
+	createdHNSEndpoint, err := hcsshim.HNSEndpointRequest("POST", "", string(hnsConfigBytes))
+	if err != nil {
+		return nil, err
+	}
+	logrus.Infof("Created HNS endpoint with ID: %v", createdHNSEndpoint.Id)
+	return createdHNSEndpoint, nil
+}
+
+// containerHotAttachEndpoint attaches the given endpoint to a running container
+func containerHotAttachEndpoint(existingHNSEndpoint *hcsshim.HNSEndpoint, containerID string) error {
+	logrus.Infof("Attaching endpoint %v to container %v", existingHNSEndpoint.Id, containerID)
+	if err := hcsshim.HotAttachEndpoint(containerID, existingHNSEndpoint.Id); err != nil {
+		logrus.Infof("Error attaching the endpoint to the container, error: %v", err)
+		return err
+	}
+	return nil
+}
+
+// deleteHNSEndpoint deletes the given endpoint if it exists
+func deleteHNSEndpoint(endpointName string) error {
+	logrus.Infof("Deleting HNS endpoint: %v", endpointName)
+	// The HNS endpoint must be manually deleted
+	hnsEndpoint, err := hcsshim.GetHNSEndpointByName(endpointName)
+	if err == nil {
+		logrus.Infof("Fetched endpoint: %v", endpointName)
+		// Endpoint exists, try to delete it
+		_, err = hnsEndpoint.Delete()
+		if err != nil {
+			logrus.Warningf("Failed to delete HNS endpoint: %q", err)
+		} else {
+			logrus.Infof("HNS endpoint successfully deleted: %q", endpointName)
+		}
+	} else {
+		logrus.Infof("No endpoint with name %v was found, error %v", endpointName, err)
+	}
+	// Return the error in case it failed, we don't want to leak any HNS Endpoints
+	return err
+}
+
+// ConfigureInterface sets up the container interface
+// Small note on this, the call to this function should be idempotent on Windows.
+// The fact that CNI add should be idempotent on Windows is stated here:
+// https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/network/cni/cni_windows.go#L38
+func ConfigureInterface(args *skel.CmdArgs, namespace string, podName string, macAddress string, ipAddress string, gatewayIP string) ([]*current.Interface, error) {
+	if strings.HasPrefix(args.Netns, containerNetNS) || strings.Compare(args.Netns, noNameNetNS) != 0 {
+		// If it is a normal container from the pod, there is nothing to do.
+		// Also, if it is not an infra container, nothing to do in this case as well.
+		logrus.Infof("CNI called for normal container or infra container, nothing to do")
+		return []*current.Interface{}, nil
+	}
+
+	ipAddr, ipNet, err := net.ParseCIDR(ipAddress)
+	if err != nil {
```

Comment:
Since we are at the early stages of development, mind adding a log message for this also?

Reply:
Doesn't really make sense for that, ipAddress is set by the OVN Watcher and will always have the format x.x.x.x/x. If this will break here then it will also break on Linux.